Src/DataTypes/Data.cs

```diff
@@ -24,5 +38,28 @@ public abstract class Data
         public abstract void UpdateTiles(List<Tile> tiles);
 
         public abstract void Cleanup();
+
+        public abstract bool Exists();
+
+        public static Data CreateDatasource(string type, string path, int batchSize)
+        {
+            string s3Url = Configuration.Instance.GetConfiguration("S3", "url");
+            string bucket = Configuration.Instance.GetConfiguration("S3", "bucket");
+
+            Data data;
+            switch (type.ToLower())
+            {
+                case "gpkg":
+                    data = new Gpkg(path, batchSize);
+                    break;
+                case "s3":
+                    data = new S3(s3Url, bucket, path, batchSize);
+                    break;
+                default:
+                    return null;
```

Comment:
throwing error seams better for easy error handling

Reply:
Fixed